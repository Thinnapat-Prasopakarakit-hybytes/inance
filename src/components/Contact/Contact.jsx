import React, { useEffect, useRef } from "react";
import "./Contact.scss";
const Contact = () => {
  return (
    <section className="contact_section layout_padding">
      <div className="container">
        <div className="heading_container">
          <h2>Contact Us</h2>
        </div>
        <div className="row">
          <div className="col-md-6">
            <form action="">
              <div>
                <input type="text" placeholder="Name" />
              </div>
              <div>
                <input type="text" placeholder="Phone Number" />
              </div>
              <div>
                <input type="email" placeholder="Email" />
              </div>
              <div>
                <input type="text" class="message-box" placeholder="Message" />
              </div>
              <div className="d-flex">
                <button>SEND</button>
              </div>
            </form>
          </div>
          <div class="col-md-6">
            <div class="map_container">
              <div class="map"></div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Contact;
