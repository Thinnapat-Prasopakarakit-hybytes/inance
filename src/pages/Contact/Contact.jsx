import React from "react";
import ContactComponent from "../../components/ContactComponent/ContactComponent";
import { APIProvider } from "@vis.gl/react-google-maps";

const Contact = () => {
  return (
    <APIProvider apiKey={process.env.REACT_APP_GOOGLE_MAPS_API_KEY}>
      <ContactComponent />
    </APIProvider>
  );
};

export default Contact;
