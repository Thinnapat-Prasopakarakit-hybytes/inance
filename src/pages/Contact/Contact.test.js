import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Contact from "./Contact";
import { APIProvider } from "@vis.gl/react-google-maps";

jest.mock("@vis.gl/react-google-maps", () => ({
  APIProvider: ({ children }) => (
    <div data-testid="mock-api-provider">{children}</div>
  ),
  Map: ({ defaultZoom, defaultCenter, gestureHandling, disableDefaultUI }) => (
    <div
      data-testid="mock-map"
      data-props={JSON.stringify({
        defaultZoom,
        defaultCenter,
        gestureHandling,
        disableDefaultUI,
      })}
    />
  ),
}));
const renderContact = () => {
  return render(
    <BrowserRouter>
      <APIProvider apiKey={process.env.REACT_APP_GOOGLE_MAPS_API_KEY}>
        <Contact />
      </APIProvider>
    </BrowserRouter>
  );
};

describe("Contact Page", () => {
  it("should render contact form", () => {
    renderContact();
    expect(screen.getByText("Contact Us")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Name")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Phone Number")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Email")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Message")).toBeInTheDocument();
  });

  it("should render map component", () => {
    renderContact();
    const mapElement = screen.getByTestId("mock-map");
    expect(mapElement).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = renderContact();
    expect(asFragment()).toMatchSnapshot();
  });
});
