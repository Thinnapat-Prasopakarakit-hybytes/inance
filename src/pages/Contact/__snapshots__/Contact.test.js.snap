// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="mock-api-provider"
  >
    <div
      data-testid="mock-api-provider"
    >
      <section
        class="contact_section layout_padding"
      >
        <div
          class="container"
        >
          <div
            class="heading_container"
          >
            <h2>
              Contact Us
            </h2>
          </div>
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <form>
                <div>
                  <input
                    name="name"
                    placeholder="Name"
                    type="text"
                  />
                </div>
                <div>
                  <input
                    name="phone"
                    placeholder="Phone Number"
                    type="text"
                  />
                </div>
                <div>
                  <input
                    name="email"
                    placeholder="Email"
                    type="email"
                  />
                </div>
                <div>
                  <input
                    class="message-box"
                    name="message"
                    placeholder="Message"
                    type="text"
                  />
                </div>
                <div
                  class="d-flex"
                >
                  <button
                    type="submit"
                  >
                    SEND
                  </button>
                </div>
              </form>
            </div>
            <div
              class="col-md-6"
            >
              <div
                class="map_container"
              >
                <div
                  data-props="{\\"defaultZoom\\":18,\\"defaultCenter\\":{\\"lat\\":40.712775,\\"lng\\":-74.005973},\\"gestureHandling\\":\\"greedy\\",\\"disableDefaultUI\\":true}"
                  data-testid="mock-map"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
